app/jobs/regular/update_username.rb

```diff
@@ -48,8 +48,7 @@ def update_posts
 
       Post
         .with_deleted
-        .joins(mentioned("posts.id"))
-        .where("a.user_id = :user_id", user_id: @user_id)
+        .where("raw ILIKE ?", "%@#{@old_username}%")
```

Comment:
We may be loading and instantiating posts that don't contain mentions here. These will left untouched in the actual mention update. 

It was either the fuzzy pattern matching above or the  regex below. Pattern matching better performance-wise?? 
https://github.com/discourse/discourse/blob/587e9ed9ba13eca315ace5e4b67489fbbb10d455/app/jobs/regular/update_username.rb#L16-L27

Reply:
`ILIKE` is fine and should be faster than regex matching. We need to execute the regex in Ruby anyway, so there's no need to run a complicated regex twice (DB and Ruby).